### PR TITLE
Revert the DcaExtractor changes

### LINKED ISSUE #4458: Tables are created if driver is not set in DCA
### Affected version(s)

4.13 - maybe also 4.9?

### Description

Since Contao 4.13 tables are also created if the driver is not set in the DCA.
According to @dmolineus, this is caused by https://github.com/contao/contao/pull/4285

Example: You define additional fields in an extension for the `tl_faq` but in a project you don't install the faq bundle. Then the table `tl_faq` is also created - although the dc driver is not defined!
see also: https://github.com/hofff/contao-social-tags/issues/25

This is a BC break, isn't it?

### PR DESCRIPTION
**Pull request by fritzmg**
Fixes #4458 

See https://github.com/contao/contao/issues/4458#issuecomment-1092826944

The changes introduced by https://github.com/contao/contao/pull/4233 & https://github.com/contao/contao/pull/4285 caused a lot of BC breaks that ultimately aren't worth it, just to support the original use case - i.e. being able to create a table without any keys. Which in the end is not a super valid use case anyway.

This PR reverts the relevant changes and reinstates the original behaviour completely.

### CODE REVIEW

**Comment by m-vo** [comment]
👍 We should IMHO keep/adjust it for 5, though instead of removing.

**Comment by fritzmg** [comment]
> 👍 We should IMHO keep/adjust it for 5, though instead of removing.

Well, I think first we would need to come up with a clear specification :)

**Comment by fritzmg** [comment]
I also added a simple test - and fixed a PHP 8 issue in the DcaExtractor.

**Comment by fritzmg** [comment]
Added more tests so that we can at least lock down the status quo.

File: core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php [hunk 1]
```diff
@@ -386,13 +386,13 @@ protected function createExtract()
...
-		if (is_a($GLOBALS['TL_DCA'][$this->strTable]['config']['dataContainer'], DC_File::class, true))
+		if (is_a($GLOBALS['TL_DCA'][$this->strTable]['config']['dataContainer'] ?? '', DC_File::class, true))
 		{
 			return;
 		}
 
 		// Return if the DC type is "Folder" and the DC is not database assisted
-		if (is_a($GLOBALS['TL_DCA'][$this->strTable]['config']['dataContainer'], DC_Folder::class, true) && empty($GLOBALS['TL_DCA'][$this->strTable]['config']['databaseAssisted']))
+		if (is_a($GLOBALS['TL_DCA'][$this->strTable]['config']['dataContainer'] ?? '', DC_Folder::class, true) && empty($GLOBALS['TL_DCA'][$this->strTable]['config']['databaseAssisted']))
 		{
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php:396]: Shouldn't this use the DataContainer method to find the class from string?

**Review by bytehead — DISMISSED** [review]

**Comment by leofeyer** [comment]
@fritzmg Can you please fix the CI chain?

**Comment by fritzmg** [comment]
@leofeyer I fixed the Code style - however there are still unhandled self deprecations left in the tests which are not related to this PR. These need to be fixed elsewhere.

**Comment by fritzmg** [comment]
The self deprecations were caused by https://github.com/contao/contao/pull/4481 - I have added a fix in this PR (7faacb1).

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**